### PR TITLE
Adding error vars names check

### PR DESCRIPTION
**Pull request by jespino**

### CODE REVIEW

File: errorVarsName/errorVars.go [hunk 1]
```diff
@@ -0,0 +1,68 @@
...
+				for idx, lhsExpr := range x.Lhs {
+					idnt, ok := lhsExpr.(*ast.Ident)
+					if !ok {
+						return true
+					}
+
+					if strings.HasPrefix(idnt.Name, "err") {
+						if len(x.Rhs) == 1 {
+							if typeAndValue, ok := pass.TypesInfo.Types[x.Rhs[0]]; ok {
+								returnTypes := strings.Split(strings.Trim(typeAndValue.Type.String(), "()"), ", ")
```
Comment (mgdelacroix) [inline errorVarsName/errorVars.go]: Same comment request that was done in #16

File: errorVarsName/errorVars.go [hunk 2]
```diff
@@ -0,0 +1,68 @@
...
+								if len(returnTypes) > idx && returnTypes[idx] == appErrorString {
+									pass.Reportf(x.Pos(), "assigning a *model.AppError variable to a `err` prefixed variable, please use appErr prefixed variable name instead.")
```
Comment (mgdelacroix) [inline errorVarsName/errorVars.go]: nit for consistency

```suggestion
									pass.Reportf(x.Pos(), "assigning a *model.AppError variable to a `err` prefixed variable, please use `appErr` prefixed variable name instead.")
```

File: errorVarsName/errorVars.go [hunk 3]
```diff
@@ -0,0 +1,68 @@
...
+								}
+							}
+						} else if len(x.Rhs) == len(x.Lhs) {
+							if typeAndValue, ok := pass.TypesInfo.Types[x.Rhs[idx]]; ok && typeAndValue.Type.String() == appErrorString {
+								pass.Reportf(x.Pos(), "assigning a *model.AppError variable to a `err` prefixed variable, please use appErr prefixed variable name instead.")
```
Comment (mgdelacroix) [inline errorVarsName/errorVars.go]:
```suggestion
								pass.Reportf(x.Pos(), "assigning a *model.AppError variable to a `err` prefixed variable, please use `appErr` prefixed variable name instead.")
```

File: errorVarsName/errorVars.go [hunk 4]
```diff
@@ -0,0 +1,68 @@
...
+							}
+						}
+					} else if strings.HasPrefix(idnt.Name, "appErr") {
+						if len(x.Rhs) == 1 {
+							if typeAndValue, ok := pass.TypesInfo.Types[x.Rhs[0]]; ok {
+								returnTypes := strings.Split(strings.Trim(typeAndValue.Type.String(), "()"), ", ")
+								if len(returnTypes) > idx && returnTypes[idx] == "error" {
+									pass.Reportf(x.Pos(), "assigning a error variable to an `appErr` prefixed variable, please use err prefixed variable name instead.")
```
Comment (mgdelacroix) [inline errorVarsName/errorVars.go]:
```suggestion
									pass.Reportf(x.Pos(), "assigning a error variable to an `appErr` prefixed variable, please use `err` prefixed variable name instead.")
```

File: errorVarsName/errorVars.go [hunk 5]
```diff
@@ -0,0 +1,68 @@
...
+								}
+							}
+						} else if len(x.Rhs) == len(x.Lhs) {
+							if typeAndValue, ok := pass.TypesInfo.Types[x.Rhs[idx]]; ok && typeAndValue.Type.String() == "error" {
+								pass.Reportf(x.Pos(), "assigning a *model.AppError variable to a `err` prefixed variable, please use appErr variable name instead.")
```
Comment (mgdelacroix) [inline errorVarsName/errorVars.go]: error message corresponds to the other case

```suggestion
								pass.Reportf(x.Pos(), "assigning a error variable to an `appErr` prefixed variable, please use `err` prefixed variable name instead.")
```

**Review by agnivade — APPROVED** [review]
Nothing to add on top of Miguel's suggestions. Good to go after those are taken care of.

Please also send a PR in the style guide.

File: errorVarsName/errorVars.go (hunk 1 above)
Comment (jespino) [reply errorVarsName/errorVars.go]: Done

**Comment by jespino** [comment]
@mgdelacroix PTAL

**Comment by jespino** [comment]
@agnivade added it here: https://github.com/mattermost/mattermost-developer-documentation/pull/771

**Review by mgdelacroix — APPROVED** [review]

**Pull request merged**